src/main/kotlin/org/rust/ide/refactoring/convertStruct/RsConvertToTupleProcessor.kt

```diff
@@ -110,8 +112,16 @@ class RsConvertToTupleProcessor(project: Project, val element: RsFieldsOwner, va
 
         val newTuplePsiElement = rsPsiFactory.createStruct("struct A$types;")
 
-        element.blockFields!!.replace(newTuplePsiElement.tupleFields!!)
-        (element as? RsStructItem)?.addAfter(rsPsiFactory.createSemicolon(), element.tupleFields!!)
+        element.blockFields?.replace(newTuplePsiElement.tupleFields!!)
+        if (element is RsStructItem) {
+            var anchor: PsiElement? = element.tupleFields
+            val whereClause = element.whereClause
+            if (whereClause != null) {
+                anchor = element.addAfter(whereClause, element.tupleFields)
+                whereClause.deleteWithSurroundingWhitespace()
+            }
+            element.addAfter(rsPsiFactory.createSemicolon(), anchor)
+        }
```

Comment:
If `where` clause is multiline, we need to keep `where` on new line. For that reason probably it is worth to keep where clause as-is and instead move block/tuple fields, something like this:

```kotlin
val blockFields = element.blockFields ?: return
val tupleFields = newTuplePsiElement.tupleFields ?: return
val whereClause = (element as? RsStructItem)?.whereClause
if (whereClause == null) {
    element.blockFields!!.replace(tupleFields)
} else {
    element.addAfter(tupleFields, whereClause.getPrevNonWhitespaceSibling())
    (blockFields.prevSibling as? PsiWhiteSpace)?.delete()
    blockFields.delete()
    whereClause.lastChild.takeIf { it.elementType == COMMA }?.delete()
}
if (element is RsStructItem) element.addAfter(rsPsiFactory.createSemicolon(), element.lastChild)
```

Reply:
Awesome. Thanks!